### PR TITLE
Initial checkin of new elasticsearch deployment tools.

### PR DESCRIPTION
**Pull request by mbookman**

### CODE REVIEW

File: elasticsearch/es-config.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+  print("Getting elasticsearch loadbalancer IP address")
+
+  # Right after deployment, the service will not yet be defined
+  # Allow for multiple attemps.
+
+  attempt_max = 12
+  for attempt in range(attempt_max):
+    # NOTE: The hard-coded service name "elasticsearch-es-http" likely
+    # limits deployments to one per cloud project. This limitation
```
Comment (wnojopra) [inline elasticsearch/es-config.py:129]: I'm not sure this is true. One cloud project can have multiple k8s clusters. You would need to have your kubectl context set to the correct cluster, each of which has its own service named elasticsearch-es-http.

We had this [setup-gcloud.sh](https://github.com/DataBiosphere/data-explorer-indexers/blob/master/util/setup-gcloud.sh) script that sets the kubectl context to point to the most recently created cluster.

Comment (mbookman) [reply elasticsearch/es-config.py:129]: I think you are right. Removed the comment.

File: elasticsearch/es-config.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  data_pool_config = {
+    'name': 'data-pool',
+    'count': config['data']['count'],
+    'machine-type': _machine_type(config['data']),
+    'node-type': 'es-data'
+  }
+
+  # The gcloud create cluster command doesn't allow one to name the default
+  # pool (which we'd like to use for the master), though there is a feature
+  # request filed for this.
```
[the remark this answers is not in the document]
Comment (mbookman) [reply elasticsearch/es-config.py:84]: Sent you the details.

File: elasticsearch/es-config.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    return int(float(ram_string[:-2]) * 1024)
+  else:
+    gen_util.err_exit(1, [
+      "Invalid 'ram' value: '{ram_string}'",
+      "Must end in 'GB'"])
+
+
+def _machine_type(node_config):
+  cpu = node_config.get('vm_cpu', 2)
+  mem = _vm_memory(node_config.get('vm_ram', '6gb'))
```
[the remark this answers is not in the document]
Comment (mbookman) [reply elasticsearch/es-config.py:50]: Thanks. Updated.

**Pull request merged**